rs/tests/test_canisters/signer/src/main.rs

```diff
@@ -14,79 +14,92 @@ use ic_signer::{GenEcdsaParams, GenSchnorrParams, GenVetkdParams};
 
 /// Generates a dummy ECDSA signature of given size parameters.
 /// The call does not verify the signature, it only generates it.
-#[update]
+#[update(manual_reply = true)]
 pub async fn gen_ecdsa_sig(
     GenEcdsaParams {
         derivation_path_length,
         derivation_path_element_size,
         key_id,
     }: GenEcdsaParams,
-) -> Result<SignWithEcdsaResponse, String> {
+) -> ManualReply<SignWithEcdsaResponse> {
     let signature_request = SignWithEcdsaArgument {
         message_hash: vec![1; 32],
         derivation_path: vec![vec![2; derivation_path_element_size]; derivation_path_length],
         key_id,
     };
 
-    sign_with_ecdsa(signature_request)
+    match sign_with_ecdsa(signature_request)
         .await
         .map(|(res,)| res)
         .map_err(|err| err.1)
+    {
+        Ok(sig) => ManualReply::one(sig),
+        Err(err) => ManualReply::reject(err),
+    }
 }
 
 /// Generates a dummy Schnorr signature of given size parameters.
 /// The call does not verify the signature, it only generates it.
-#[update]
+#[update(manual_reply = true)]
 pub async fn gen_schnorr_sig(
     GenSchnorrParams {
         message_size,
         derivation_path_length,
         derivation_path_element_size,
         key_id,
     }: GenSchnorrParams,
-) -> Result<SignWithSchnorrResponse, String> {
+) -> ManualReply<SignWithSchnorrResponse> {
     let signature_request = SignWithSchnorrArgument {
         message: vec![1; message_size],
         derivation_path: vec![vec![2; derivation_path_element_size]; derivation_path_length],
         key_id,
     };
 
-    sign_with_schnorr(signature_request)
+    match sign_with_schnorr(signature_request)
         .await
         .map(|(res,)| res)
         .map_err(|err| err.1)
+    {
+        Ok(sig) => ManualReply::one(sig),
+        Err(err) => ManualReply::reject(err),
+    }
```

Comment:
Does this work?
```suggestion
    match sign_with_schnorr(signature_request).await {
        Ok((sig,)) => ManualReply::one(sig),
        Err((_, err)) => ManualReply::reject(err),
    }
```
The same in the other places

Reply:
It does! This is definitely better, thanks